lib/job.js

```diff
@@ -11,7 +11,10 @@ function Job(queue, jobId, data, options) {
   this.id = jobId;
   this.progress = 0;
   this.data = data || {};
-  this.options = options || {};
+  this.options = options || {
+    timestamp: Date.now(),
+    stacktraces: []
+  };
```

Comment:
Seems like we'd want to set these regardless of the `options` passed in, no? That way we avoid a surprise when we go to `unshift` onto the `stacktraces` array only to discover it was never set.